### PR TITLE
✨ Update to CRD v1

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**:
- Updates the CRDs to v1.
- Updates kubebuilder-tools to 1.16.4
- Updates capi to v0.3.0-rc.0

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
Update to CRD v1 
```

### CODE REVIEW

File: scripts/fetch_ext_bins.sh [hunk 1]
```diff
@@ -26,7 +26,7 @@ if [[ -n "${TRACE}" ]]; then
   set -x
 fi
 
-k8s_version=1.14.1
+k8s_version=1.16.7
```
Comment (detiber) [inline scripts/fetch_ext_bins.sh]: 1.16.4 should work here. It is what is currently in use by CAPI and a WIP PR for CAPA

**Comment by nader-ziada** [comment]
I think you also have to set the `preserveUnknownFields` to false

**Comment by richardcase** [comment]
> I think you also have to set the `preserveUnknownFields` to false

I found that i needed that with my image change: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/fb24a3e49e6a6126b829a3a0946fa7efaab31686/Makefile#L175

File: go.mod [hunk 2]
```diff
@@ -12,21 +12,20 @@ require (
...
-	k8s.io/api v0.17.1
-	k8s.io/apiextensions-apiserver v0.17.1 // indirect
-	k8s.io/apimachinery v0.17.1
-	k8s.io/client-go v0.17.1
+	k8s.io/api v0.17.2
+	k8s.io/apimachinery v0.17.2
+	k8s.io/client-go v0.17.2
 	k8s.io/klog v1.0.0
 	k8s.io/utils v0.0.0-20191114184206-e782cd3c129f
 	sigs.k8s.io/cluster-api v0.2.6-0.20200106222425-660e6b945a27
```
Comment (nader-ziada) [inline go.mod]: we can now bump this to v0.3.0-rc.0

File: Makefile [hunk 3]
```diff
@@ -172,7 +172,7 @@ generate-go: $(CONTROLLER_GEN) $(MOCKGEN) $(CONVERSION_GEN) ## Runs Go related g
 generate-manifests: $(CONTROLLER_GEN) ## Generate manifests e.g. CRD, RBAC etc.
 	$(CONTROLLER_GEN) \
 		paths=./api/... \
```
Comment (nader-ziada) [inline Makefile]: we also need `crd:preserveUnknownFields=false` here

Comment (CecileRobertMichon) [reply Makefile]: I'm not sure we do, @vincepri actually removed `crd:preserveUnknownFields=false` in his PR  https://github.com/kubernetes-sigs/cluster-api/pull/2132/files#diff-b67911656ef5d18c4ae36cb6741b7965L263, I think that was required before this change (and that's why it was needed in v1alpha2 and in both the webhook PRs).

Comment (vincepri) [reply Makefile]: We don't need the flag for CRDv1, it's assumed to be false. `crd:preserveUnknownFields` needs to be set only for v1beta1 CRDs.

File: go.mod (hunk 2 above)
Comment (CecileRobertMichon) [reply go.mod]: I'll make that change in a separate PR

**Comment by CecileRobertMichon** [comment]
/hold

We need to wait until capi v0.3.0-rc.1 drops

**Pull request merged**